src/utils/string_utils.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @internal
+ * Removes leading zeros and explicit plus from textual representation of a number.
+ */
+export function removeLeadingZerosAndExplicitPlus(str: string): string {
+  if (str === '') {
+    return str;
+  }
+
+  let startIndex = 0;
+
+  const isNegative = str[startIndex] === '-';
+  const isExplicitlyPositive = str[startIndex] === '+';
+
+  if (isExplicitlyPositive || isNegative) {
+    startIndex += 1;
+  }
+
+  let foundInsignificantZero = false;
+
+  while (str[startIndex] === '0') {
```

Comment:
I guess this works because indexing a string with a length that's too long just returns undefined so we break from the loop but relying on behavior such as this feels odd.  Could we add a length check and convert this to a for loop?

```typescript
for (; startIndex < str.length && str[startIndex] === '0'; ++startIndex) {
...
}
```

Reply:
good catch! just converted it